securedrop_client/logic.py

```diff
@@ -412,8 +421,17 @@ def on_sync_success(self) -> None:
 
     def on_sync_failure(self, result: Exception) -> None:
         """
-        Called when syncronisation of data via the API queue fails.
+        Called when syncronisation of data via the API fails. Just log and try again since this is
+        a background task.
+        """
+        logger.debug('The SecureDrop server cannot be reached due to Error: {}'.format(result))
+        self.sync_api()
```

Comment:
I can see this makes sense if the reason for the sync errors is that `ApiInaccessibleError` is being raised - when you investigated, _was_ it `ApiInaccessibleError` that was the cause? If it is, I'm not following why the `ApiInaccessibleError` is raised. My expectation is that it should only occur when one of the following is true:

* User's token has expired
* User's token is somewhere otherwise invalid
* There is a bug

Reply:
When we run a sync, we kick off a `MetadataSyncJob` which on success will kick off `MessageDownloadJob`s, `ReplyDownloadJob`s, or `FileDownloadJob`s. On failure (`ApiInaccessibleError` or `RequestTimeoutError` or any kind of failure) we used to _always_ show "The SecureDrop server cannot be reached" error with the Retry link. Now, on failure, we check if the sync ran because of a user action (clicking refresh) or not. If the action was not from a user action, we don't ask the user to automatically retry.

Looking at this more, I'm realizing that I forgot to add the code that checks if the exception in that comes back via the failure signal of `MetadataSyncJob` is because we could not reach the server. My change introduces retrying for any kind of error which is problematic.

Overall, I'm curious if you think it's a good approach to periodically retry connecting to the server when it cannot be reached until we can successfully metadata sync. Let's discuss this more because I may be going in a direction you disagree with.

Reply:
> when you investigated, was it ApiInaccessibleError that was the cause?

When I investigated I saw that for any kind of error during sync_api we would show "The SecureDrop server cannot be reached" error with the Retry link. So the answer is sometimes.

One thing we still need to add is increasing the timeouts for all jobs before we pause the queue in case 5 seconds is not long enough to download messages, metadata for file downloads, etc.

Reply:
so in latest `master`, calling `sync_api` is now just going to add a new `MetadataSyncJob` to the queue, it won't restart the queue, and since the queue will be paused if a `ApiInaccessibleError` or `RequestTimeoutError` occurred, that job will get automatically retried upon queue restart, so I don't _think_ we need this logic. Let me know if you disagree. If the sync job fails for some other reason, it means something else is wrong (like https://github.com/freedomofpress/securedrop-proxy/issues/13).

> I'm curious if you think it's a good approach to periodically retry connecting to the server when it cannot be reached until we can successfully metadata sync.

I do think we want this but I don't think we want to do this as part of this ticket (I think that's #491) and I do also agree with the direction of eventually using the MetadataSyncJob for #491.

Reply:
you are right, resuming queue is what was intended, just fixed this.